Mathlib/AlgebraicGeometry/OpenImmersion.lean

```diff
@@ -826,4 +826,58 @@ def Scheme.openCoverOfSuprEqTop {s : Type*} (X : Scheme) (U : s → Opens X)
     exact (Opens.mem_iSup.mp this).choose_spec
 #align algebraic_geometry.Scheme.open_cover_of_supr_eq_top AlgebraicGeometry.Scheme.openCoverOfSuprEqTop
 
+section category
+
+/--
+A morphism between open covers `𝓤 ⟶ 𝓥` indicates that `𝓤` is a refinement of `𝓥`.
+Since open covers of schemes are indexed, the definition also involves a map on the
+indexing types.
+-/
+structure Scheme.OpenCover.Hom {X : Scheme.{u}} (𝓤 𝓥 : Scheme.OpenCover.{v} X) where
+  /-- The map on indexing types associated to a morphism of open covers. -/
+  idx : 𝓤.J → 𝓥.J
```

Comment:
Would `mapIndex` be a better name for `idx`?

Reply:
I can change it if you absolutely insist, but I would prefer to leave it as `idx` for the following reasons:
1. It's a three-letter abbreviation just like `obj`, `app`, which appear elsewhere in the category theory library.
2. The abbreviation `idx` already appears in various places in the more "programmy" side of lean.
3. `mapIndex` is just too long in my opinion.